src/server/scripts/Spells/spell_quest.cpp

```diff
@@ -2453,6 +2453,39 @@ class spell_q12414_hand_over_reins : public SpellScriptLoader
         }
 };
 
+// 13790 13793 13811 13814 - Among the Champions
+// 13665 13745 13750 13756 13761 13767 13772 13777 13782  13787 - The Grand Melee
+class spell_q13665_q13790_bested_trigger : public SpellScriptLoader
+{
+    public:
+        spell_q13665_q13790_bested_trigger() : SpellScriptLoader("spell_q13665_q13790_bested_trigger") { }
+
+        class spell_q13665_q13790_bested_trigger_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_q13665_q13790_bested_trigger_SpellScript);
+
+            void HandleScript(SpellEffIndex /*effIndex*/)
+            {
+                if (Unit* target = GetHitUnit())
+                {
+                    if (target->IsVehicle())
+                        target = target->GetOwner();
```

Comment:
you could use `GetCharmerOrOwnerOrSelf` here
that will prevent target is null and isvehicle is not needed